Add data type support page (#1772)
ROCm/hipBLASLt: docs/reference/data-type-support.rst

.. meta::
   :description: hipBLASLt API library data type support
   :keywords: hipBLASLt, ROCm, API library, API reference, data type, support

.. _data-type-support:

******************************************
Data type support
******************************************

> This section lists the supported data types for the GEMM operation of hipBLASLt, 

Reviewer: ```suggestion
This topic lists the supported data types for the hipBLASLt GEMM operation, 
```

which is performed by :ref:`hipblasltmatmul`. Here is the equation:

.. math::

 D = Activation(alpha \cdot op(A) \cdot op(B) + beta \cdot op(C) + bias)

Where *op( )* refers to in-place operations, such as transpose and

> non-transpose, and *alpha* and *beta* are scalars.

Reviewer: Confirm that these variables should be in italics and not code ticks. Use code ticks if they are actual variable names.

Author: Fixed.

.. note:: 
  

>   The ``__hip_fp8_e4m3_fnuz`` and ``__hip_fp8_e5m2_fnuz`` are only supported on

Reviewer: ```suggestion
  The ``__hip_fp8_e4m3_fnuz`` and ``__hip_fp8_e5m2_fnuz`` data types are only supported on
```

.. meta::
   :description: hipBLASLt library data type support
   :keywords: hipBLASLt, ROCm, data type support

.. _data-type-support:

******************************************
Data type support
******************************************

This topic lists the supported data types for the hipBLASLt GEMM operation, 
which is performed by :ref:`hipblasltmatmul`. Here is the equation:

.. math::

 D = Activation(alpha \cdot op(A) \cdot op(B) + beta \cdot op(C) + bias)

Where :math:`op( )` refers to in-place operations, such as transpose and
non-transpose, and :math:`alpha` and :math:`beta` are scalars.

.. note:: 
  

>   The ``__hip_fp8_e4m3_fnuz`` and ``__hip_fp8_e5m2_fnuz`` data types are only

Reviewer 2: The actual data type used by user are hipblaslt_f8_fnuz and hipblaslt_bf8_fnuz.
Any reason that you use __hip_fp8_e4m3_fnuz and __hip_fp8_e5m2_fnuz?
And should we have a table to match data type to HIP enum data type? example: HIP_R_8F_E4M3_FNUZ

Author: Hi @jichangjichang 

This was an excellent question, so I will put that table to match data type to HIP enum data type to [main data type support page](https://rocm.docs.amd.com/en/latest/reference/precision-support.html). Outside BLAS libraries, the library developers are using the HIP types and I wanted to uniform that. Maybe I will use the enums on the data type page. Let's make first the main page changes.

Author: @jichangjichang I added the compute modes and hipBLASLt type list. It can be, that I shared modes, which are not supported, please check it. 

Also removed the HIP types, and now I am adding the HIP enum data type map to the main data type support page.

.. meta::
   :description: hipBLASLt library data type support
   :keywords: hipBLASLt, ROCm, data type support

.. _data-type-support:

******************************************
Data type support
******************************************

This topic lists the supported data types for the hipBLASLt GEMM operation, 
which is performed by :ref:`hipblasltmatmul`. 

The ``hipDataType`` enumeration defines data precision types and is primarily 
used when the data reference itself does not include type information, such as
in ``void*`` pointers. This enumeration is mainly utilized in BLAS libraries.

> The hipBLASLt input and output types listed in the following table.

Reviewer: ```suggestion
The hipBLASLt input and output types are listed in the following table.
```

.. list-table::
    :header-rows: 1

    *
      - hipDataType
      - hipBLASLt type
      - Description

    * 
      - ``HIP_R_8I``
      - ``hipblasLtInt8``
      - 8-bit real signed integer.

    * 
      - ``HIP_R_32I``
      - ``hipblasLtInt32``
      - 32-bit real signed integer.

    * 
      - ``HIP_R_8F_E4M3_FNUZ``
      - ``hipblaslt_f8_fnuz``
      - 8-bit real float8 precision floating-point

    * 
      - ``HIP_R_8F_E5M2_FNUZ``
      - ``hipblaslt_bf8_fnuz``
      - 8-bit real bfloat8 precision floating-point

    * 
      - ``HIP_R_16F``
      - ``hipblasLtHalf``
      - 16-bit real half precision floating-point

    * 
      - ``HIP_R_16BF``
      - ``hipblasLtBfloat16``
      - 16-bit real bfloat16 precision floating-point

    * 
      - ``HIP_R_32F``
      - ``hipblasLtFloat``
      - 32-bit real single precision floating-point

> The hipBLASLt compute modes listed in the following table.

Reviewer: ```suggestion
The hipBLASLt compute modes are listed in the following table.
```
